tests/checkpointing/test_model_checkpoint.py

```diff
@@ -367,7 +367,7 @@ def test_model_checkpoint_topk_all(tmpdir):
     seed_everything(1000)
     epochs = 2
     model = EvalModelTemplate()
-    checkpoint_callback = ModelCheckpoint(dirpath=tmpdir, monitor="early_stop_on", save_top_k=-1)
+    checkpoint_callback = ModelCheckpoint(dirpath=tmpdir, monitor="some_val", save_top_k=-1)
```

Comment:
Why did you modify some tests ?

Reply:
@tchaton  That test was failing on macos and windows (often) with PyTorch 1.7
but it is passing locally with 1.7. Tried with some other monitor and test god passed.
So requested for a review for a while.

Reply:
I will have a look at that test but it makes sense to me this test starts failing, because the last checkpoint is not guaranteed to be the best. the monitor key should correspond to the training- or validation loss, not early stopping.

Reply:
But early_stop_on is defined in validation_epoch_end

Reply:
Ahh, tests are producing different best_model_path
https://github.com/PyTorchLightning/pytorch-lightning/actions/runs/333879691

Reply:
yes it seems the change in pytorch version produces slightly different results which leads to different best model path... it's a bit strange.

Reply:
Any findings/updates here @ydcjeff? Don't mind assisting here to find the source of this issue (we probably shouldn't merge till we know why this happens)

Reply:
Thank you @SeanNaren 
[this](https://github.com/PyTorchLightning/pytorch-lightning/pull/3821/checks?check_run_id=1327084401) and [this result](https://github.com/PyTorchLightning/pytorch-lightning/pull/3821/checks?check_run_id=1327084438) are different although we are doing same seed with different pytorch versions.

The comparison is early_stop_on values.
How should I approach this? The ModelCheckpoint is saving the best model tho.

Reply:
@ydcjeff Were you able to reproduce this locally? I wasn't able to...

Reply:
No, I can't reproduce the CI errors. I also have one machine.
The results are always same on same machines, different on different machines and pytorch versions.
And that's normal
See: https://discuss.pytorch.org/t/reproducibility-over-different-machines/63047/2

Reply:
I have windows here and PT 1.7 and the test runs, it must have something todo with the platform and the random number generator.

@ydcjeff increased the epochs in a previous commit https://github.com/PyTorchLightning/pytorch-lightning/pull/3821/commits/9c649c725a376037464c3fabfe1ce38cf47a861d
and it showed the best model according to the validation loss was not the last, which would mean this example is either overfitting or not learning at all, leading to some random checkpoint to be the best.

Reply:
Yep, the model is kinda overfitting, the only way I can think of now is tracking with `monitor`